redisearch/src/main/java/io/github/dengliming/redismodule/redisearch/protocol/decoder/SearchResultDecoder.java

```diff
@@ -30,24 +31,57 @@
  */
 public class SearchResultDecoder implements MultiDecoder<SearchResult> {
 
-    @Override
-    public Decoder<Object> getDecoder(int paramNum, State state) {
-        return null;
-    }
-
-    @Override
-    public SearchResult decode(List<Object> parts, State state) {
-        Long total = (Long) parts.get(0);
-        boolean noContent = total.longValue() == parts.size() + 1;
-        List<Document> documents = new ArrayList<>(total.intValue());
-        for (int i = 1; i < parts.size(); i++) {
-            if (noContent) {
-                documents.add(new Document((String) parts.get(i), 1.0d, null));
-            }
-            else if ((i + 1) % 2 != 0) {
-                documents.add(new Document((String) parts.get(i - 1), 1.0d, (Map<String, Object>) parts.get(i)));
-            }
-        }
-        return new SearchResult(total, documents);
-    }
+	@Override
+	public Decoder<Object> getDecoder(int paramNum, State state) {
+		return null;
+	}
+
+	@Override
+	public SearchResult decode(List<Object> parts, State state) {
+
+		int documentSize = 0;
+		Long total = (Long) parts.get(0);
+		boolean noContent = total == parts.size() + 1;
+
+		// Calculates Document Size searching for the first LinkedHashMap on parts...
+		// It would be better if I could access the commands parameters to check if we are dealing with the Score values...
```

Comment:
Why not?

```
// io.github.dengliming.redismodule.redisearch.protocol.RedisCommands
RedisCommand FT_SEARCH = new RedisCommand<>("FT.SEARCH", new ListMultiDecoder2(new SearchResultDecoder(), new StringMapInfoDecoder()));

RedisCommand FT_SEARCH_WITH_SCORES = new RedisCommand<>("FT.SEARCH", new ListMultiDecoder2(new SearchResultDecoder(true), new StringMapInfoDecoder()));

// io.github.dengliming.redismodule.redisearch.protocol.decoder.SearchResultDecoder
public SearchResultDecoder(boolean withScores) {
}


// io.github.dengliming.redismodule.redisearch.RediSearch
return commandExecutor.readAsync(getName(), StringCodec.INSTANCE, searchOptions.isWithScores ? FT_SEARCH_WITH_SCORES : FT_SEARCH, args.toArray());
```
Is it just what you want?

Reply:
Yes, it's definitely a good approach, but I was worried about creating a new RedisCommand as you said. I'll make the changes and commit it.